.pre-commit-config.yaml

```
@@ -24,7 +24,7 @@ repos:
       - id: cargo-check
         args: ['--manifest-path', './dask_planner/Cargo.toml', '--verbose', '--']
       - id: clippy
-        args: ['--manifest-path', './dask_planner/Cargo.toml', '--verbose', '--']
+        args: ['--manifest-path', './dask_planner/Cargo.toml', '--verbose', '--', '-D', 'warnings']
```

Just want to note that we can also configure `clippy` to automatically fix some of the more easily corrected warnings with

```suggestion
        args: ['--manifest-path', './dask_planner/Cargo.toml', '--fix', '--allow-dirty', '--verbose', '--', '-D', 'warnings']
```

Was just hesitant to do this by default as the error message when leaving out `--allow-dirty` seemed to imply that this could result in destructive changes; @andygrove @jdye64 do you have any preferences here?

---

I would prefer that we fix any warnings ourselves, even if we are often just copying & pasting Clippy's suggestions. I think this is an excellent way to learn more about writing idiomatic Rust code.

---

Agree with Andy's comments here